dask/dataframe/tests/test_groupby.py

```diff
@@ -1883,6 +1883,16 @@ def test_std_object_dtype(func):
     assert_eq(func(df), func(ddf))
 
 
+def test_std_columns_int():
+    # Make sure std() works when index_by is a df with integer column names
+    # Non regression test for issue #3560
+
+    df = pd.DataFrame({0: [5], 1: [5]})
+    ddf = dd.from_pandas(df, npartitions=2)
+    by = dask.array.from_array([0, 1]).to_dask_dataframe()
+    ddf.groupby(by).std()
```

Comment:
I'm not sure this test is valid. The group key should I think be the same length as `df`.

Can you try something like

```
df = pd.DataFrame({0: [0, 1], 1: [1, 2]})
ddf = dd.from_pandas(df, npartitoins=2)
by = pd.Series([0, 1])

expected = df.groupby(by).std()
result = ddf.groupby(dd.from_pandas(by, 2)).std()
assert_eq(result, expected)
```

Reply:
I can confirm that the current test fails on master with the same error as in the original issue.

To make the suggested test pass I need to set the following args:

`assert_eq(result, expected, check_meta=False, equal_nan=True)`

All entries are Nans so I'm not sure that makes  lot of sense to compare the results? Let me know which one you prefer.

Reply:
Thanks. I suppose it's all NaN because the Dataframe is small, so the std calculation returns NaN? Can you make a larger dataframe?

Reply:
That said, we may be going past the original issue now. Remind me, that issue was specifically about column names, which this test exercises?

Reply:
yes the original issue was about column name mismatch, not about expected results. Do you still need me to adapt your test?

Reply:
Nope, I think we're good. Sorry about the misunderstanding.